contracts/sol6/reserves/KyberFprReserveV2.sol

```diff
@@ -0,0 +1,389 @@
+            if (rate > sanityRate) return 0;
+        }
+
+        return rate;
+    }
+
+    function isAddressApprovedForWithdrawal(IERC20 token, address addr)
+        external
+        view
+        returns (bool)
+    {
+        return approvedWithdrawAddresses[keccak256(abi.encodePacked(address(token), addr))];
+    }
+
+    function tradeEnabled() external view returns (bool) {
+        return configData.tradeEnabled;
+    }
+
+    function maxGasPriceWei() external view returns (uint128) {
+        return configData.maxGasPriceWei;
+    }
+
+    function doRateValidation() external view returns (bool) {
+        return configData.doRateValidation;
+    }
+
+    /// @dev return available balance of a token that reserve can use
+    ///      if using weth, call getBalance(eth) will return weth balance
+    ///      if using wallet for token, will return min of balance and allowance
+    /// @param token token to get available balance that reserve can use
+    function getBalance(IERC20 token) public view returns (uint256) {
+        address wallet = getTokenWallet(token);
+        IERC20 usingToken;
+
+        if (token == ETH_TOKEN_ADDRESS) {
+            if (wallet == address(this)) {
+                // reserve should be using eth instead of weth
+                return address(this).balance;
+            }
+            // reserve is using weth instead of eth
+            usingToken = weth;
+        } else {
+            if (wallet == address(this)) {
+                // not set token wallet or reserve is the token wallet, no need to check allowance
+                return token.balanceOf(address(this));
+            }
+            usingToken = token;
+        }
+
+        uint256 balanceOfWallet = usingToken.balanceOf(wallet);
+        uint256 allowanceOfWallet = usingToken.allowance(wallet, address(this));
+
+        return minOf(balanceOfWallet, allowanceOfWallet);
+    }
+
+    /// @dev return wallet that holds the token
+    ///      if token is ETH, check tokenWallet of WETH instead
+    ///      if wallet is 0x0, consider as this reserve address
+    function getTokenWallet(IERC20 token) public view returns (address wallet) {
+        wallet = (token == ETH_TOKEN_ADDRESS)
+            ? tokenWallet[address(weth)]
+            : tokenWallet[address(token)];
+        if (wallet == address(0)) {
+            wallet = address(this);
+        }
+    }
+
+    /// @dev do a trade, re-validate the conversion rate, remove trust assumption with network
+    /// @param srcToken Src token
+    /// @param srcAmount Amount of src token
+    /// @param destToken Destination token
+    /// @param destAddress Destination address to send tokens to
+    /// @param doRateValidation do rate validation or not
+    function doTrade(
+        IERC20 srcToken,
+        uint256 srcAmount,
+        IERC20 destToken,
+        address payable destAddress,
+        uint256 conversionRate,
+        bool doRateValidation
```

Comment:
compiler warning



contracts/sol6/reserves/KyberFprReserveV2.sol:314:9: Warning: This declaration shadows an existing declaration.
        bool doRateValidation
        ^-------------------^
contracts/sol6/reserves/KyberFprReserveV2.sol:257:5: The shadowed declaration is here:
    function doRateValidation() external view returns (bool) {
    ^ (Relevant source part starts here and spans across multiple lines).

Reply:
change to `validateRate`